src/test/java/org/opensearch/neuralsearch/query/AgenticSearchQueryIT.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.query;
+
+import org.junit.After;
+import org.junit.Before;
+import org.apache.lucene.tests.util.LuceneTestCase.AwaitsFix;
+import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+
+@AwaitsFix(bugUrl = "Ignoring until we find a way to fetch access, secret key for remote model")
+public class AgenticSearchQueryIT extends BaseNeuralSearchIT {
+
+    private static final String TEST_INDEX = "test-agentic-index";
+    private static String TEST_AGENT_ID;
+    private static String TEST_MODEL_ID;
+    private static final String TEST_QUERY_TEXT = "Find documents about machine learning";
+    private static final String AWS_ACCESS_KEY_ID = System.getenv("AWS_ACCESS_KEY_ID");
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        updateClusterSettings();
+        if (TEST_AGENT_ID == null) {
+            try {
+                String connectorId = createTestConnector();
```

Comment:
when the AWS_ACCESS_KEY_ID is null, please skip the IT, because when it runs in local clusters, it cannot read the variables from github, this file will have AWS_ACCESS_KEY_ID is null

Reply:
right just point out the difference. because running on github and local cluster would get different variables value assigned.